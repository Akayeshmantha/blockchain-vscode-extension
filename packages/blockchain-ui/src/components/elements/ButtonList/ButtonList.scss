.bx--tab-content {
    width: 100%;
    padding-left: 0px;
    background-color: transparent;
}