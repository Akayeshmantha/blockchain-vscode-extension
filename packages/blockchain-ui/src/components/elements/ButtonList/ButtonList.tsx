import React, { Component } from 'react';
import './ButtonList.scss';
import { Button } from 'carbon-components-react';

interface IProps {
    buttons: any[];
}

class ButtonList extends Component<IProps> {

    // This element will probably just be an array of button elements.
    // Will then loop through the array and render the buttons.
    // E.g. this could have 'Back', 'Next' buttons inline next to each other.

    // This should also make it easy to add/remove buttons from this list.
    // E.g. Remove 'Next' button and add 'Finish' button.

    render(): JSX.Element {

        const buttonList: JSX.Element[] = [];
        for (const button of this.props.buttons) {
            buttonList.push(<Button kind={button.kind} disabled={button.disabled}>{button.label}</Button>);
        }

        return (
            <div className='bx--btn-set'>
                {buttonList}
            </div>
        );
    }
}

export default ButtonList;
