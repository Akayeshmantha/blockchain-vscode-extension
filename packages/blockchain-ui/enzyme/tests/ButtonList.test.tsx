import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import ButtonList from '../../src/components/elements/ButtonList/ButtonList';

chai.should();
chai.use(sinonChai);

describe('ButtonList component', () => {
    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<ButtonList buttons={[]}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render button', () => {
        const component: ReactWrapper<{buttons: any[]}, {}> = mount(<ButtonList buttons={[{label: 'Next', kind: 'primary', disabled: false}]}/>);
        component.text().should.equal('Next');
    });
});
