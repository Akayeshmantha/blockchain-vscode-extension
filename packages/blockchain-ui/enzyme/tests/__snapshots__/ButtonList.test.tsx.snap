// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonList component should render the expected snapshot 1`] = `
<div
  className="bx--btn-set"
/>
`;
